### DISCUSSION TITLE
Allow PRs to be generated against different base branch

### DISCUSSION THREAD

**Discussion by robaca** [renovatebot/renovate#25668] [Suggest an Idea]
### Tell us more.

In our Monorepo, we have a default `develop` branch, which runs the mainline pipeline, and a `main` branch that contains the latest stable state. For all our development, we always first branch from `main`, but later file PRs against `develop`, so that during development, we always are sure to act on a working base.

It would be great to support this in renovate by allowing to configure a different PR base branch ("target branch").

We currently workaround this by a workflow step `gh pr edit --base develop`, but in that case the behavior of renovate is not clearly defined (does it handle merge conflicts and rebasing correctly etc).

**Comment by rarkins** [comment]
Renovate config always must be present in the default branch, buy you can configure the target branch(es) for PRs using the `baseBranches` config option

**Reply by robaca** [reply]
@rarkins This works, and we have `baseBranches: ['main']` configured. But I want to configure renovate to branch away from `main`, but create PRs for these branches against our default branch `develop` as described above.

**Reply by rarkins** [reply]
Unfortunately not supported and I'm not sure if it could be added elegantly either due to the way we calculate whether a branch is modified or not

**Reply by robaca** [reply]
Doing it by hand without direct support has another drawback: At the moment where we retarget the PR, renovate starts to ignore it and creates a second PR for the same branch, targeting to `main` again. Also, closing a retargeted PR does not cause Renovate to ignore that dependency, so we get a new PR on next run.

**Reply by rarkins** [reply]
One of the reasons I'm against this is that I assume it will increase complexity, decrease maintainability, and only for a very small subsection of users (possibly just you, for now). I don't mind if you explore the code the create a draft PR or branch to _demonstrate_ it can be done elegantly and we can reconsider, but you should definitely run against your branch/PR for some time to ensure there's no gotchas you missed.

**Reply by robaca** [reply]
After some more investigation, we decided to let renovate branch from `develop` instead of `main`.

**Reply by bzakdd** [reply]
Just for the record, I would also benefit from this feature.
